samples/geotag.cpp

```diff
@@ -556,26 +495,24 @@ bool sina(const char* s, const char** a) {
     const char* A = a[i];
     while (*A == '-')
       A++;
-    bResult = stricmp(s, A) == 0;
+    bResult = _stricmp(s, A) == 0;
     i++;
   }
   return bResult;
 }
 
-int readFile(const char* path, const Options& /* options */) {
+int readFile(const fs::path& path, const Options& /* options */) {
   if (!fs::exists(path)) {
     return typeUnknown;
   }
 
-  const char* ext = strstr(path, ".");
-  if (ext) {
-    const char* docs[] = {".doc", ".txt", nullptr};
-    const char* code[] = {".cpp", ".h", ".pl", ".py", ".pyc", nullptr};
-    if (sina(ext, docs))
+  for (auto doc : {".doc", ".txt"})
+    if (path.extension() == doc)
       return typeDoc;
-    if (sina(ext, code))
+
+  for (auto code : {".cpp", ".h", ".pl", ".py", ".pyc"})
```

Comment:
```suggestion
  for (const auto& code : {".cpp", ".h", ".pl", ".py", ".pyc"})
```

Reply:
I don’t think there’s a benefit here. const char* is trivially copyable.